src/sparseml/exporters/transforms/base_transform.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import Any
+
+
+class BaseTransform(ABC):
+    """
+    Interface that all transforms must implement.
+    A transform is an object that, when applied to a model,
+    does a single modification to it
+    and returns a modified model.
+    """
+
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def __call__(self, model: Any) -> Any:
```

Comment:
It felt appropriate to apply the transform through the `__call__` method, feel free to challenge me.

Reply:
prefer to have an explicitly named method like `apply()` as well, can have `__call__` just got into `apply`

akin to `__call__` and `forward()` in pytorch